deepchecks/vision/batch_wrapper.py

```diff
@@ -94,6 +96,23 @@ def __len__(self):
         dataset = self._context.get_data_by_kind(self._dataset_kind)
         return len(list(dataset.data_loader.batch_sampler)[self.batch_index])
 
+    def vision_properties(self, raw_data: List, properties_list: List[Dict], input_type: PropertiesInputType):
+        """Calculate and cache the properties for the batch according to the propety input type"""
+        properties_list = validate_properties(properties_list)
+        # if there are no cached properties at all, calculate all the properties on the list,
+        # else calculate only those that were not yet calculated.
+        dataset = self._context.get_data_by_kind(self._dataset_kind)
+        if self._vision_properties_cache[input_type.value] is None:
+            self._vision_properties_cache[input_type.value] = dataset.calc_properties(raw_data, properties_list)
+        else:
+            properties_to_calc = [p for p in properties_list if p['name'] not in
```

Comment:
Not an important comment, but sounds like you can use only this line if you initialized _vision_properties_cache to be an empty dict per input type rather than None